bundles/org.eclipse.packagedrone.utils.rpm/src/org/eclipse/packagedrone/utils/rpm/build/RpmBuilder.java

```diff
@@ -139,6 +142,89 @@ public String toString ()
         }
     }
 
+    public static class Feature
+    {
+        private String name;
+
+        private String evr;
+
+        private EnumSet<RpmDependencyFlags> flags;
+
+        private String description;
+
+        public Feature ()
+        {
+
+        }
+
+        public Feature ( String name, String evr, EnumSet<RpmDependencyFlags> flags, String description )
+        {
+            this.name = name;
+            this.evr = evr;
+            this.flags = flags;
+            this.description = description;
+        }
+
+        public String getName ()
+        {
+            return name;
+        }
+
+        public void setName ( String name )
+        {
+            this.name = name;
+        }
+
+        public String getEvr ()
+        {
+            return evr;
+        }
+
+        public void setEvr ( String evr )
+        {
+            this.evr = evr;
+        }
+
+        public EnumSet<RpmDependencyFlags> getFlags ()
+        {
+            return flags;
+        }
+
+        public void setFlags( EnumSet<RpmDependencyFlags> flags )
+        {
+            this.flags = flags;
+        }
+
+        public String getDescription ()
+        {
+            return description;
+        }
+
+        public void setDescription( String description )
+        {
+            this.description = description;
+        }
+    }
+
+    public static List<Feature> features = new ArrayList<> ();
```

Comment:
I don't understand why this is needed, and why this is a public, static, mutable field.

It is ok to have this just for debugging/information purposes. But in this case I would think it would be best to make this immutable by not setting an initializer value, then building a new List in the static initializer section, and setting the static field by:

~~~java
features = Collections.unmodifiableList(newFeatures);
~~~

Reply:
There is an initial set of features, and then some features may be added (depending on the chosen compression or hash algorithm), so it cannot be immutable as written.

However, the public modifier seems to be a mistake, and I agree that I don't think that this field should be directly modifiable.